src/test/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinitionTest.java

```diff
@@ -181,4 +181,22 @@
         r.assertLogContains("version one", r.assertBuildStatusSuccess(p.scheduleBuild2(0, new ParametersAction(new StringParameterValue("VERSION", "one")))));
     }
 
+    @Issue("JENKINS-42836")
+    @Test public void usingParameterInScriptPath() throws Exception {
+        sampleRepo.init();
+        sampleRepo.write("flow.groovy", "echo 'version one'");
+        sampleRepo.git("add", "flow.groovy");
+        sampleRepo.write("otherFlow.groovy", "echo 'version two'");
+        sampleRepo.git("add", "otherFlow.groovy");
+        sampleRepo.git("commit", "--all", "--message=commits");
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        CpsScmFlowDefinition def = new CpsScmFlowDefinition(new GitSCM(Collections.singletonList(new UserRemoteConfig(sampleRepo.fileUrl(), null, null, null)),
+                Collections.singletonList(new BranchSpec("master")),
+                false, Collections.<SubmoduleConfig>emptyList(), null, null, Collections.<GitSCMExtension>emptyList()), "${SCRIPT_PATH}");
+        def.setLightweight(false); // TODO SCMFileSystem.of cannot pick up build parameters
```

Comment:
No no, this is wrong, just copied-pasted from the previous test which dealt with SCM configuration. You _are_ expanding `scriptPath` even in lightweight mode. So either delete this line, or test it in each mode for full coverage.

Reply:
whoops, good catch. Thanks!